server/src/main/java/org/elasticsearch/common/ReferenceDocs.java

```diff
@@ -89,7 +87,18 @@ public enum ReferenceDocs {
     // this comment keeps the ';' on the next line so every entry above has a trailing ',' which makes the diff for adding new links cleaner
     ;
 
-    private static final Map<String, String> linksBySymbol;
+    private static final Map<String, LinkComponents> linksBySymbol;
+
+    record LinkComponents(String path, String fragment) {
+        static LinkComponents ofLink(String link) {
```

Comment:
optional nit: should we add an assertion here to check the link doesn't already contain a query part?

Reply:
Thanks 👍